builder/dockerfile/imagecontext.go

```diff
@@ -110,16 +121,24 @@ type imageMount struct {
 	id        string
 	ctx       builder.Context
 	release   func() error
-	ic        *imageContexts
 	runConfig *container.Config
+	mounter   builder.ImageMounter
+}
+
+func newImageMount(mounter builder.ImageMounter) *imageMount {
+	return &imageMount{mounter: mounter}
+}
+
+func newImageMountWithID(id string, mounter builder.ImageMounter) *imageMount {
+	return &imageMount{id: id, mounter: mounter}
 }
 
 func (im *imageMount) context() (builder.Context, error) {
 	if im.ctx == nil {
 		if im.id == "" {
 			return nil, errors.Errorf("could not copy from empty context")
 		}
-		p, release, err := im.ic.b.docker.MountImage(im.id)
+		p, release, err := im.mounter.MountImage(im.id)
```

Comment:
Some of this refactor was to enable this change. So that this function doesn't have to reach two levels into a dependency (ref: https://en.wikipedia.org/wiki/Law_of_Demeter)